src/main.cpp

```diff
@@ -650,6 +650,13 @@ int main ( int argc, char** argv )
             exit ( 1 );
         }
 
+        if ( strServerInfo.split ( ";" ).count() < 3 )
```

Comment:
if no cli argument is given, I think we shouldn't output anything. I'd argue that's the point of this PR.

Reply:
No - it's an error.  The argument is _mandatory_, it's a command line syntax error.  It won't get this far.

Reply:
The problem comes if you swap `-s` and `--serverinfo`:
```
$ ./Jamulus -p 50000 --serverinfo -s
- Starting in server mode by default (due to compile time option)
- selected port number: 50000
- server info: -s
No GUI support compiled. Running in headless mode.
"--serverinfo '-s'": must contain [name];[city];[country] - value will be ignored
... continues to run ...
```
It could error out.

Reply:
Of course, the above is when running compiled server only.  Without that, it's still an error:
````
p$ ./Jamulus -p 50000 --serverinfo -s
- selected port number: 50000
- server info: -s
./Jamulus: Server only option(s) '--serverinfo' used.  Did you omit '--server'?
```

Reply:
This is why I say the `main.cpp` validation needs tightening up.  But not in this PR -- it's a much bigger problem than this one parameter.

Reply:
Hmm. So you say `jamulus -s` is invalid? 

The point of this PR is that jamulus -s doesn't give an error. `jamulus -s --serverinfo` should give an error.

Reply:
No, `Jamulus -s` is perfectly valid.  The issue is that the parser incorrectly accepts `-s` as the argument to `--serverinfo` and swallows it, then spits out the `--serverinfo` as it's a server-only parameter (and it's not running as a server).  And _that's_ inconsistent, depending on whether you compiled server-only or not.  So it's a mess bigger than the scope of this PR.

Reply:
Ah ok. That's a separate issue. However I get a warning of invalid serverinfo for `jamulus -s` that should not happen.

Reply:
Give the full command line, otherwise I can't comment.

Reply:
Ah, right, I see...  Oops.

Reply:
Patch applied (and one that should probably be used when validating all the other parameters).